### PR TITLE
Update the explanation of the features (variable time resolution)

### LINKED ISSUE #256: Update formulation in the documentation with the ranges/matrices
### What and Why

The current mathematical formulation in the documentation is according to temporal handling in the model.

### Possible Drawbacks

The main drawback is that it will be more challenging to follow the mathematical formulation of Tulipa. Therefore, we should have a formulation in the documentation that is the simplest possible. In other words, we keep the formulation in the documentation as a `vanilla version` to keep it readable and simple and add sections for the temporal structure and other more complex topics that require a certain level of abstraction.

@g-moralesespana, @abelsiqueira, and @clizbe What do you think?

### Related Issues

_No response_

### PR DESCRIPTION
**Pull request by datejada**
# Pull request details

## Describe the changes made in this pull request

- Updates the documentation of the variable time resolution
- Update in the variable time resolution to cover more cases in the example

## List of related issues or pull requests

Closes #256

## Collaboration confirmation

As a contributor I confirm

- [x] I read and followed the instructions in README.dev.md
- [x] The documentation is up to date with the changes introduced in this Pull Request (or NA)
- [x] Tests are passing
- [x] Lint is passing

### CODE REVIEW

**Comment by datejada** [comment]
Thanks! @clizbe, I changed the test case to cover one extra case we discussed with @g-moralesespana. So, the documentation and the test case match one to the other.

File: docs/src/features.md [hunk 1]
```diff
@@ -1,151 +1,192 @@
...
 
-This section explains the main features in the optimization model inside [TulipaEnergyModel.jl](https://github.com/TulipaEnergy/TulipaEnergyModel.jl).
+*TulipaEnergyModel.jl* incorporates two fundamental concepts that serve as the foundation of the optimization model:
+
+- **Energy Assets**: representation of a physical asset that can produce, consume, store, balance, or convert energy. Some examples of what these assets can represent are:
+  - Producer: wind turbine, solar panel
+  - Consumer: electricity demand, heat demand
+  - Storage: battery, pumped-hydro storage
+  - Balancing Hub: electricity network that serves as a connection among other energy assets
+  - Conversion: power plants, electrolizers
```
Comment (gnawin) [inline docs/src/features.md:10]: electrolizers -> electrolyzers?

File: docs/src/features.md [hunk 2]
```diff
@@ -1,151 +1,192 @@
...
+- The `phs` is in a six-hour resolution represented by the range `1:6`, meaning the storage balance is determined every six hours.
+- The flow from the phs to the balance (`phs,balance`) represents the discharge of the `phs`. This flow has a variable resolution of two blocks from hour 1 to 4 (i.e., `1:4`) and from hour 5 to 6 (i.e., `5:6`), which differs from the one defined for the charging flow from the `wind` asset.
+- The `demand` consumption has hourly input data with one connection to the `balance` hub:
+  - The flow from the balance hub to the demand (`balance,demand`) has a uniform resolution of three hours; therefore, it has two blocks, one from hour 1 to 3 (i.e., `1:3`) and the other from hour 4 to 6 (i.e., `4:6`).
+- The `balance` hub integrates all the different assets with their different resolutions. The highest resolution of all connections determines the balance equation for this asset. Therefore, the resulting resolution is into two blocks, one from hour 1 to 4 (i.e., `1:4`) and the other from hour 5 to 6 (i.e., `5:6`).
```
Comment (gnawin) [inline docs/src/features.md:43]: highest -> lowest?

**Comment by gnawin** [comment]
Hi @datejada! I'm reading this part and have some minor comments...I still have a few other questions for my understanding though but we'd better discuss

**Pull request merged**